keyless/circuit-data/templates/helpers/misc.circom

```diff
@@ -112,6 +112,37 @@ template EmailVerifiedCheck(maxEVNameLen, maxEVValueLen, maxUIDNameLen) {
 }
 
 
+template StringBodies(len) {
+  signal input in[len];
+  signal output out[len];
+
+
+  signal quotes[len];
+  signal quote_parity[len];
+  signal quote_parity_1[len];
+  signal quote_parity_2[len];
+
+  quotes[0] <== IsEqual()([in[0], 34]); 
+  quote_parity[0] <== IsEqual()([in[0], 34]); 
+
+  for (var i = 1; i < len; i++) {
+    var is_quote = IsEqual()([in[i], 34]); 
+    var prev_is_backslash = IsEqual()([in[i-1], 92]); 
+    quotes[i] <== is_quote * (1 - prev_is_backslash);
+    quote_parity_1[i] <== quotes[i] * (1 - quote_parity[i-1]);
+    quote_parity_2[i] <== (1 - quotes[i]) * quote_parity[i-1];
+    quote_parity[i] <== quote_parity_1[i] + quote_parity_2[i];
+  }
```

Comment:
can this recognize `"abc\\"` which means string `abc\`?

Reply:
You're right. [Fixed](https://github.com/aptos-labs/aptos-core/pull/13069/commits/92d1965210c0a46b9cdefab1bff99781b8a789c8).